### PR TITLE
feat: added iupd record

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Added record for handle iupd generation.

#### Motivation and Context
The record has been implemented to centralize the iupd generation logic

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/aca/domain/Iupd.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package it.pagopa.aca.domain
+
+@JvmRecord
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/aca/domain/Iupd.kt:3]: why a java record here? kotlin data class should just be good here

Comment (Fernando-Granato) [reply src/main/kotlin/it/pagopa/aca/domain/Iupd.kt:3]: https://kotlinlang.org/docs/jvm-records.html#requirements 🙂 tell me what you think

Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/aca/domain/Iupd.kt:3]: Yes, my question is, why use a java record? a kotlin [data class](https://kotlinlang.org/docs/data-classes.html) should model this entity

**Review by pietro-tota — APPROVED** [review]

File: src/main/kotlin/it/pagopa/aca/domain/Iupd.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    fun value(): String {
+        return StringBuilder()
+            .append("ACA")
+            .append("_")
+            .append(fiscalCode)
+            .append("_")
+            .append(iuv)
+            .toString()
+    }
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/aca/domain/Iupd.kt:12]:
```suggestion
    fun value(): String = "ACA_${fiscalCode}_$iuv"
```
🚲

**Pull request merged**